src/Sylius/Bundle/CoreBundle/Resources/config/handlers.xml

```diff
@@ -24,6 +24,27 @@
                 </argument>
             </call>
         </service>
+
+        <service id="sylius.handler.currency_change" class="Sylius\Component\Core\Handler\CompositeCodeChangeHandler">
```

Comment:
I understand your point now. It is obvious that it will reduce code but still i think it should be separate. We need more feedback from team. WDYT @pamil, @tuka217 , @GSadee , @Zales0123 , @lchrusciel.